File: monitor.py
import requests
import json
import os
from datetime import datetime
from pathlib import Path

def load_config():
    with open('config.json', 'r') as f:
        return json.load(f)

def send_telegram_message(bot_token, chat_id, message):
    url = f"https://api.telegram.org/bot{bot_token}/sendMessage"
    payload = {
        "chat_id": chat_id,
        "text": message,
        "parse_mode": "HTML"
    }
    try:
        response = requests.post(url, json=payload, timeout=10)
        response.raise_for_status()
    except Exception as e:
        print(f"Failed to send Telegram message: {e}")

def check_url(url, timeout=10):
    try:
        response = requests.get(url, timeout=timeout)
        return response.status_code == 200, response.status_code
    except requests.exceptions.RequestException as e:
        return False, str(e)

def save_history(results):
    history_file = Path('docs/history.json')
    history_file.parent.mkdir(exist_ok=True)
    
    history = []
    if history_file.exists():
        with open(history_file, 'r') as f:
            history = json.load(f)
    
    history.append(results)
    
    # Keep last 1000 checks
    history = history[-1000:]
    
    with open(history_file, 'w') as f:
        json.dump(history, f, indent=2)

def main():
    config = load_config()
    bot_token = os.environ.get('TELEGRAM_BOT_TOKEN')
    chat_id = os.environ.get('TELEGRAM_CHAT_ID')
    
    if not bot_token or not chat_id:
        print("Error: TELEGRAM_BOT_TOKEN and TELEGRAM_CHAT_ID must be set")
        return
    
    timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
    failed_urls = []
    results = {
        "timestamp": timestamp,
        "checks": []
    }
    
    for url in config['urls']:
        is_up, status = check_url(url)
        results["checks"].append({
            "url": url,
            "status": "up" if is_up else "down",
            "code": status if isinstance(status, int) else None,
            "error": status if isinstance(status, str) else None
        })
        
        if is_up:
            print(f"✓ {url} - OK")
        else:
            print(f"✗ {url} - FAILED ({status})")
            failed_urls.append((url, status))
    
    save_history(results)
    
    if failed_urls:
        message = f"🚨 <b>Sentinel Alert</b>\n\n"
        message += f"Time: {timestamp} UTC\n\n"
        for url, status in failed_urls:
            message += f"❌ {url}\nStatus: {status}\n\n"
        
        send_telegram_message(bot_token, chat_id, message)

if __name__ == "__main__":
    main()

File: pages/api/monitor.js
import axios from 'axios';
import fs from 'fs';
import path from 'path';

const config = {
  urls: [
    "https://vultrconf.com",
    "https://vultrconf.com/telecommunications",
    "https://vultrconf.com/hospitality", 
    "https://vultrconf.com/finance",
    "https://vultrconf.com/healthcare",
    "https://bogus-url.com"
  ]
};

async function sendTelegramMessage(botToken, chatId, message) {
  try {
    await axios.post(`https://api.telegram.org/bot${botToken}/sendMessage`, {
      chat_id: chatId,
      text: message,
      parse_mode: "HTML"
    });
  } catch (error) {
    console.error('Failed to send Telegram message:', error.message);
  }
}

async function checkUrl(url, timeout = 10000) {
  try {
    const response = await axios.get(url, { timeout });
    return { isUp: response.status === 200, status: response.status };
  } catch (error) {
    return { isUp: false, status: error.message };
  }
}

function getTimezoneTime() {
  const now = new Date();
  
  // IST (UTC+5:30)
  const istTime = new Date(now.getTime() + (5.5 * 60 * 60 * 1000));
  
  // US Eastern (UTC-5 or UTC-4 depending on DST)
  const usTime = new Date(now.toLocaleString("en-US", {timeZone: "America/New_York"}));
  
  return {
    ist: istTime.toISOString().slice(0, 19).replace('T', ' '),
    us: usTime.toISOString().slice(0, 19).replace('T', ' ')
  };
}

export default async function handler(req, res) {
  const botToken = process.env.TELEGRAM_BOT_TOKEN;
  const chatId = process.env.TELEGRAM_CHAT_ID;
  
  if (!botToken || !chatId) {
    return res.status(500).json({ error: 'Missing Telegram credentials' });
  }
  
  const times = getTimezoneTime();
  const failedUrls = [];
  const results = {
    timestamp: new Date().toISOString(),
    checks: []
  };
  
  // Check all URLs
  for (const url of config.urls) {
    const { isUp, status } = await checkUrl(url);
    
    results.checks.push({
      url,
      status: isUp ? 'up' : 'down',
      code: typeof status === 'number' ? status : null,
      error: typeof status === 'string' ? status : null
    });
    
    if (!isUp) {
      failedUrls.push({ url, status });
    }
  }
  
  // Save history (in production, you'd use a database)
  // For now, we'll just log it
  console.log('Monitoring results:', results);
  
  // Send notifications for failures
  if (failedUrls.length > 0) {
    let message = "🚨 <b>Sentinel Alert</b>\n\n";
    message += `🇮🇳 IST: ${times.ist}\n`;
    message += `🇺🇸 EST: ${times.us}\n\n`;
    
    for (const { url, status } of failedUrls) {
      message += `❌ ${url}\nStatus: ${status}\n\n`;
    }
    
    await sendTelegramMessage(botToken, chatId, message);
  }
  
  res.status(200).json({ 
    success: true, 
    checked: config.urls.length,
    failed: failedUrls.length,
    results 
  });
}